script.js

```diff
@@ -1 +1,98 @@
-console.log("Assignment 2");
\ No newline at end of file
+
+var bookList;
+var bookListCurrent;
+
+async function fetchBooks(){
+    let books;
+    await fetch('book_list.json').then(response => {
+        if(!response.ok){
+            throw new Error("HTTP error " + response.status);
+        }
+        return response.json();
+    })
+    .then(data => {
+        books = data;
+    });
+    
+    return books;
+}
+
+function renderBookList(books){
+    let tBody = document.getElementById('tbody-all-books');
+    tBody.innerHTML = "";
+
+    books.forEach(function (book) {
+        tBody.insertAdjacentHTML('beforeend', `<td>${book.bookId}</td><td>${book.genre}</td><td>${book.price}</td><td>No</td>`);
```

Comment:
Minor optimisation:
We can prepare html in js var and then insert the complete html in one go.